### PR TITLE
Remove support for deprecated intent/entity format

### LINKED ISSUE #1267: Misleading warning regarding old story format
**Rasa Core version**: 0.11.*, master

**Issue**: Wrong deprecation message

This deprecation message is misleading:

https://github.com/RasaHQ/rasa_core/blob/95e430f3a6307b82cac0509f0db9d4472fd46709/rasa_core/processor.py#L236

In fact, the [interpreter checks](https://github.com/RasaHQ/rasa_core/blob/95e430f3a6307b82cac0509f0db9d4472fd46709/rasa_core/interpreter.py#L180) for brackets, not for leading `_`. The check was updated but the deprecation message wasn't.

### PR DESCRIPTION
**Pull request by MetcalfeTom**
Fixes #1267
Nothing to add in the documentation

**Proposed changes**:
- Remove all checks for the format
- Remove all functions which supported use of the old format, now rasa_core will simply error instead
- Remove all examples of old format in sample data and comments
- Add test for the processor parsing method

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added tests
- [x] updated the changelog

### CODE REVIEW

File: rasa_core/interpreter.py [hunk 1]
```diff
@@ -147,54 +147,11 @@ def extract_intent_and_entities(user_input):
...
         """Parse a text message."""
 
-        if self.is_using_deprecated_format(text):
-            intent, confidence, entities = \
-                self.deprecated_extraction(text)
-        else:
-            intent, confidence, entities = \
-                self.extract_intent_and_entities(text)
+        intent, confidence, entities = \
+            self.extract_intent_and_entities(text)
```
Comment (ricwo) [inline rasa_core/interpreter.py]: could make this one line now

File: rasa_core/processor.py [hunk 2]
```diff
@@ -240,18 +240,9 @@ def _get_action(self, action_name):
 
     def _parse_message(self, message):
         # for testing - you can short-cut the NLU part with a message
-        # in the format _intent[entity1=val1,entity=val2]
+        # in the format /intent{"entity1": val1, "entity2": val2}
```
Comment (ricwo) [inline rasa_core/processor.py]: 👍

File: rasa_core/processor.py [hunk 3]
```diff
@@ -240,18 +240,9 @@ def _get_action(self, action_name):
...
-                message.text.startswith("_")):
-            if RegexInterpreter.is_using_deprecated_format(message.text):
-                warnings.warn(
-                        "Parsing messages with leading `_` is deprecated and "
-                        "will be removed. Instead, prepend your intents with "
-                        "`{0}`, e.g. `{0}mood_greet` "
-                        "or `{0}restart`.".format(INTENT_MESSAGE_PREFIX))
-            parse_data = RegexInterpreter().parse(message.text)
-        else:
+        if (message.text.startswith(INTENT_MESSAGE_PREFIX)):
```
Comment (ricwo) [inline rasa_core/processor.py]: The intent-message case should still be parsed by the RegexInterpreter, no? Also, we still need the `else` case for a non-intent message

Comment (MetcalfeTom) [reply rasa_core/processor.py]: Wow, that was a pretty glaring error on my part.  I'm not sure how the tests didn't pick up on that so I will write one

File: tests/test_processor.py [hunk 4]
```diff
@@ -22,10 +23,7 @@ def test_message_processor(default_processor):
 
 
 def test_parsing(default_processor):
-    class Message():
-        text = '/greet{"name":"boy"}'
-
-    message = Message()
+    message = Message('/greet{"name":"boy"}')
```
Comment (ricwo) [inline tests/test_processor.py]: tiny thing, but the examples we're giving have `{"name": "boy"}`, so with a space between the dict's key and value

**Pull request merged**